R/data-types.R

```diff
@@ -0,0 +1,40 @@
+setGeneric("dbiDataType", function(x) standardGeneric("dbiDataType"))
+
+data_frame_data_type <- function(x) {
+  vapply(x, dbiDataType, FUN.VALUE = character(1), USE.NAMES = TRUE)
+}
+
+varchar_data_type <- function(x) {
+  paste0("VARCHAR(", max(nchar(as.character(x))), ")")
```

Comment:
I wonder if this should return "TEXT" - I don't like the idea that the type of a column depends on its specific values